### PR TITLE
chore: convert useAdminRolePermissions to TS

### PR DESCRIPTION
**Pull request by gu-stav**
### What does it do?

Convert `useAdminRolePermissions` to TS and move it into the settings NS.

### CODE REVIEW

File: packages/core/admin/shared/permissions.ts [hunk 1]
```diff
@@ -0,0 +1,52 @@
+type Condition = {
+    id: string;
+    displayName: string;
+    category: string;
+}
+
+type Permission = {
```
Comment (joshuaellis) [inline packages/core/admin/shared/permissions.ts:7]: I think a lot of these can be taken from the `@strapi/utils` package iirc.

Comment (gu-stav) [reply packages/core/admin/shared/permissions.ts:7]: Could you be a bit more specific? 🙈 I checked it, but couldn't find anything useful ...

Comment (joshuaellis) [reply packages/core/admin/shared/permissions.ts:7]: sorry i wrote the wrong package – that was dumb of me:

```ts
import type { domain } from '@strapi/permissions';

type Permission = domain.permission.Permission;
```

Comment (gu-stav) [reply packages/core/admin/shared/permissions.ts:7]: ah, oh ... a new world appears.

Comment (joshuaellis) [reply packages/core/admin/shared/permissions.ts:7]: Joe Biden GIF by GIPHY News

Comment (gu-stav) [reply packages/core/admin/shared/permissions.ts:7]: Are we ok having `@strapi/permissions` as a dependency of `strapi/admin`?

Comment (joshuaellis) [reply packages/core/admin/shared/permissions.ts:7]: We're only using types, so I would assume so, this is how i've normally gone about it.

Comment (gu-stav) [reply packages/core/admin/shared/permissions.ts:7]: Mh, so this is the old problem again: the types in the permission package represent the BE internal data-structure, but not what the admin app deals with:

```
export interface Permission {
  action: string;
  actionParameters?: Record<string, unknown>;
  subject?: string | object | null;
  properties?: object;
  conditions?: string[];
}
```

Comment (gu-stav) [reply packages/core/admin/shared/permissions.ts:7]: @joshuaellis What do you think about leaving them here for now until the controller was converted to TS to move on? I can check with Jamie next week on how to share them best according to the ADR you wrote this week...

Tests should be fixed now.

Comment (joshuaellis) [reply packages/core/admin/shared/permissions.ts:7]: yes 

Lisa Simpson Full Steam Ahead GIF by The Simpsons

**Review by joshuaellis — APPROVED** [review]

**Pull request merged**